baybe/constraints/conditions.py

```diff
@@ -134,12 +174,13 @@ def evaluate(self, data: pd.Series) -> pd.Series:  # noqa: D102
                 "This operation is error-prone and not supported. Only use threshold "
                 "conditions with numerical parameters."
             )
-        func = rpartial(_threshold_operators[self.operator], self.threshold)
-        if self.operator in _valid_tolerance_operators:
-            func = rpartial(func, atol=self.tolerance)
-
+        func = self.generate_operator_function()
         return data.apply(func)
 
+    def to_polars(self, param=None) -> Callable | pl.Expr:
+        """Convert the condition to a callable for Polars."""
+        return self.generate_operator_function()
```

Comment:
this is still ugly, a function called `to_polars` should not return `Callable`. It causes a lot of ugly patterns downstream where we use this

I remember saying you tried to change it, but what did you try out?

Here is what happens where the operator returned by this is applied:
```
op = cond.to_polars(param=None)
satisfied.append(op(pl.col(self.parameters[k]))) 
```
Looking at the leftover `params=None` in the function signature, did you try to pass the list of parameters and return this?
```
def to_polars(self, params: list[str]) -> pl.Expr:
    op = self.generate_operator_function()
    return op(pl.col(self.parameters)) 
```